src/postcodeInfoClient.ts

```diff
@@ -0,0 +1,85 @@
+import * as requestDefault from 'request'
+import * as requestPromise from 'request-promise-native'
+import { Address } from './address'
+import { Point } from './point'
+import { PostcodeInfoResponse } from './postcodeInfoResponse'
+
+export class PostcodeInfoClient {
+  constructor (private readonly apiToken: string,
+               private readonly request: requestDefault.RequestAPI<requestPromise.RequestPromise,
+                 requestPromise.RequestPromiseOptions,
+                 requestDefault.RequiredUriUrl> = requestPromise,
+               private readonly apiUrl: string = 'https://postcodeinfo.service.justice.gov.uk') {
+  }
+
+  public lookupPostcode (postcode: string): Promise<PostcodeInfoResponse> {
```

Comment:
I found this method to be quite long, what do you think about decomposing it a bit? I guess the intermediate promises can be composed within own functions and `.then` handler could have something extracted as well.

Reply:
simplified